services/src/main/scala/cromwell/services/metadata/impl/archiver/ArchiveMetadataSchedulerActor.scala

```diff
@@ -35,27 +58,62 @@ class ArchiveMetadataSchedulerActor(archiveMetadataConfig: ArchiveMetadataConfig
 
   def archiveNextWorkflow(): Future[Boolean] = {
     for {
-      maybeWorkflowId <- getNextWorkflowId()
+      maybeWorkflowId <- lookupNextWorkflowToArchive()
       result <- maybeWorkflowId match {
         case Some(id) => for {
-          _ <- streamMetadataToGcs(id)
-          _ <- markWorkflowAsArchived(id)
+          path <- Future(archiveMetadataConfig.makePath(id))
+          _ = log.info(s"Archiving metadata for $id to ${path.pathAsString}")
+          dbStream <- fetchStreamFromDatabase(id)
+          _ <- streamMetadataToGcs(path, dbStream)
+          _ <- updateMetadataArchiveStatus(id, Archived)
```

Comment:
If an error occurs inside the `streamMetadataToGcs`, shouldn't we mark the workflow with status as `ArchivedFailed`? Or is the idea that the workflows are either in `Archived` state or not, and we simply keep retrying if there are any failures?

Reply:
I think the idea was to just keep retrying, with loud warnings each time. Maybe we need a ticket to forward those alerts and warnings to slack, or somewhere we'll actually see them

Reply:
See: https://broadworkbench.atlassian.net/browse/BW-602

Reply:
Thank you for creating the ticket. I agree that we should have some kind of alert to know when archiving is failing. We can also add instrumentation around failures and have a graph in Grafana as well.